data/kahet/kahet missions.txt

```diff
@@ -65,6 +65,7 @@ mission "First Contact: Ka'het: Remnant 1A"
 		has "remnant: blood test pure"
 		not "First Contact: Ka'het: Remnant 1B: offered"
 	on offer
+		set "global: First Contact: Ka'het"
```

Comment:
Well, these conditions are for having met / discovered the faction in question. There's no requirement for having a conversation with them, or being on good terms, or any of that. Note that the conversation where the Remnant one is set the player *may* just high tail it back onto their ship and leave, never having learned anything beyond "they want to test my blood!"

That being said, this could be a candidate for an invisible mission on one of the Builder worlds, too.

Reply:
> That being said, this could be a candidate for an invisible mission on one of the Builder worlds, too.

Such a mission already exists: the one that triggers adding the Graveyard label.
The Ringworld Debris stuff could also be used.
Indeed, these conditions are what is used to trigger the missions you are setting the first contact condition in.
The label mission or a new mission with the label mission and ringworld debris thing as triggers would probably be much more appropriate than this mission, which requires you to have passed the Remnant blood test.

Reply:
That makes sense. Done.